### DISCUSSION TITLE
Boards do not show in COM list, unable to upload

### DISCUSSION THREAD

**Discussion by mrtopkid** [earlephilhower/arduino-pico#1019] [General]
Hi all,

Having a issue with uploading anything from the ide to any of my rp2040 boards (aside from one, more on that later). I'm using the newer 2.0.2 ide, 2.6.3 of the arduino-pico board lib. Initially every one of my boards worked flawlessly, issue has crept onto almost all of my boards (4 of them). 

I installed arduino-pico as per the instructions. Uploaded my first sketch with the 'Uploading the First Sketch' instructions (https://arduino-pico.readthedocs.io/en/latest/install.html#uploading-sketches). Code uploaded, but board would then not show in the COM list. I should note that, on all of these boards, this had worked at some stage in the last few days. 

Here are the things that I have tried to resolve the issue:
1. Tried to re-jig the drivers as per the instructions on the docs (https://arduino-pico.readthedocs.io/en/latest/install.html#windows-7-driver-notes)
2. Flash nuked each board and started from the 'first sketch' phase
3. Wiped Arduino completely from my system and reinstalled
4. Since experiencing the issue for the first time:  tried on a board that was brand new
5. Deleted these from devmgr and began from first sketch phase
![image](https://user-images.githubusercontent.com/78351891/204160500-f664ea01-0176-428e-b5f8-9ee6e6fb4404.png)
6. Reinstalled my OS (win 11). Had a bloated OS so was gonna do this anyway

After OS reinstall I:
1. Installed arduino
2. Installed 'arduino-pico' as per the instructions
3. Held bootsel and plugged in a board
4. Uploaded a known working sketch via Tools->Port->UF2 Board

After trying the above, the board would still not show in the COM list, and the only way to upload is via putting the board into bootsel. Furthermore, windows _being the super helpful OS it is_ showed me this upon plugging in a board:
![image](https://user-images.githubusercontent.com/78351891/204160684-5d001ca4-eeb2-4814-aa76-0f8ab55f4c9e.png)

As mentioned earlier, one of the boards does in fact work, with no issues whatsoever. I'm lead to think that this issue is one of a few things:
1. Corrupt board (I'd have thought this was unlikely, since the bootloader is untouchable, and all boards were flash nuked)
2. Windows driver issues (perhaps windows is being silly, and using different drivers for some boards? I do not know enough about this to confidently say)
3. An issue with arduino / arduino-pico

Apologies if this is the wrong place / forum to post this issue if the issue is in fact not due to this board lib. Any help would be hugely appreciated.

**Comment by Fredjikrang** [comment]
I had this same issue after updating Arduino IDE to 2.0.2 and to any of the Arduino-Pico 2.6.X versions. I downgraded Arduino-Pico to 2.5.4 and everything worked again. I haven't tried with 2.6.4 yet.

**Reply by microwinie** [reply]
I am using linux (ubuntu) as a platform.  I found a workaround: the permissions need to be changed whenever the port is created (sudo chmod 777 /dev/ttyACM0).  It's annoying because the device comes and goes, for example, once for the download and another for the SERIAL connection.  The other option would be to run the IDE as root, which isn't very attractive.
I expect this is an IDE issue -who knows?  Hope this helps

**Reply by earlephilhower** [reply]
That's probably a groups problem, @microwinie .  See https://askubuntu.com/questions/207999/how-to-add-dialup-group .  When your user is in the `dialout` group you should automatically have access to `/dev/ttyXXX`...
